karapace/protobuf/proto_file_element.py

```diff
@@ -99,8 +99,13 @@ def __eq__(self, other: "ProtoFileElement") -> bool:  # type: ignore
     def __repr__(self) -> str:
         return self.to_schema()
 
-    def compare(self, other: "ProtoFileElement", result: CompareResult) -> CompareResult:
-
+    def compare(
+        self,
+        other: "ProtoFileElement",
+        result: CompareResult,
+        self_dependency_types: Optional[List[TypeElement]] = None,
+        other_dependency_types: Optional[List[TypeElement]] = None,
+    ) -> CompareResult:
```

Comment:
We have a type for dependencies... and my code which I working on already use there exactly this type. please not that Your fixes will interfere with my code logic which I just debugging. You can see it at https://github.com/instaclustr/karapace/pull/24